src/main/java/CancelApplication/CancelApplicationDsGateway.java

```diff
@@ -0,0 +1,32 @@
+package CancelApplication;
+
+import Entities.Group;
+import Entities.User;
+
+public interface CancelApplicationDsGateway {
+    boolean groupExists(String groupname);
+    User getUser(String username);
+    Group getGroup(String groupname);
+
```

Comment:
I think one thing that we should talk about is what we should name our DsGateway methods. Of course, this gate way is specific to you, but we all use the getUser call. I personally have getUserByName(). So, it would be best to discuss what we should name shared method calls to avoid confusion.

Reply:
Yeah, that's a great point. Once we set up the repository and have it implement all our gateways -- I think it'll be very clear to see any repetitions. Of course, we could also discuss this beforehand.